ledger/src/blockstore.rs

```diff
@@ -2250,6 +2236,64 @@ impl Blockstore {
         Ok(())
     }
 
+    /// Validates compatibility between an UpdateParent and a BlockHeader ParentMeta.
+    ///
+    /// This function determines which ParentMeta is which (UpdateParent vs BlockHeader) and
+    /// validates that they are compatible. Works regardless of parameter order.
+    fn update_parent_block_header_checks(slot: Slot, x: &ParentMeta, y: &ParentMeta) -> Result<()> {
```

Comment:
maybe better named as `check_parent_meta_consistency` since you allow block components of the same type. And update doc comment